kernel/core/maths/functionimplementation.cpp

```diff
@@ -2737,6 +2737,75 @@ value_type parser_isnan(const value_type& v)
 }
 
 
+/////////////////////////////////////////////////
+/// \brief This function returns the utc offset
+///
+/// \return value_type
+///
+/////////////////////////////////////////////////
+value_type parser_get_utc_offset()
+{
+    auto now = std::chrono::system_clock::now();
+    std::time_t now_c = std::chrono::system_clock::to_time_t(now);
+    std::tm* tm_ptr = std::gmtime(&now_c);
+    int utc_offset_seconds = std::mktime(tm_ptr) - now_c;
+    int utc_offset_hours = utc_offset_seconds / 3600;
+    return utc_offset_hours;
+}
+
+
+/////////////////////////////////////////////////
+/// \brief This function returns whether the year
+/// is a leap year
+///
+/// \param nYear const value_type&
+/// \return value_type
+///
+/////////////////////////////////////////////////
+value_type parser_is_leap_year(const value_type& nYear)
+{
+    if (intCast(nYear) % 4 == 0){
+        if (intCast(nYear) % 100 != 0 || intCast(nYear) % 400 == 0) {
+            return 1.0;
+        }
+    }
+    return 0.0;
+}
+
+
+
+/////////////////////////////////////////////////
+/// \brief This function returns whether the
+/// current time is a summer time
+///
+/// \param vElements const value_type*
+/// \param nElements int
+/// \return value_type
+///
+/////////////////////////////////////////////////
+value_type parser_is_daylightsavingtime(const value_type* vElements, int nElements)
+{
+    int year = nElements > 0 ? intCast(vElements[0].real()) : 1970;
+    int month = nElements > 1 ? intCast(vElements[1].real()) : 1;
+    int day = nElements > 2 ? intCast(vElements[2].real()) : 1;
+
+    std::tm time_info = {};
+    time_info.tm_year = year - 1900;
+    time_info.tm_mon = month - 1;
+    time_info.tm_mday = day;
+    time_info.tm_hour = 12;          // Set to noon to avoid ambiguity

```

Comment:
The `time_stamp` contains all necessary time fields to fill this structure. However, you want to research how to actually achieve to handle dates before 1.1.1970. Problematic is also the small data range, because this might not work after some date in 2038 (32 bits problem).

There might be a solution within the boost library to achieve this behaviour.

Reply:
Still open, isn't it?